### ISSUE TITLE
Extbuf in try_read_command_binary function is too small

### ISSUE THREAD

**Issue by TomasKorbar** [memcached/memcached#863]
**Describe the bug**
Hi, since a upgrade of gcc in Fedora, memcached has been unable to build
and i was seeing this error in logs:
```
In function 'complete_incr_bin',
    inlined from 'dispatch_bin_command' at proto_bin.c:1010:17,
    inlined from 'try_read_command_binary' at proto_bin.c:119:9,
    inlined from 'try_read_command_binary' at proto_bin.c:52:5:
proto_bin.c:272:31: error: array subscript 'union protocol_binary_request_incr[0]' is partly outside array bounds of 'char[45]' [-Werror=array-bounds]
  272 |     req->message.body.delta = ntohll(req->message.body.delta);
      |                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
proto_bin.c: In function 'try_read_command_binary':
proto_bin.c:113:14: note: object 'extbuf' of size 45
  113 |         char extbuf[sizeof(c->binary_header) + BIN_MAX_EXTLEN+1];
      |              ^~~~~~
```

extbuf buffer is too small for protocol_binary_request_incr. This can be fixed with 
addition of 3 more so the final size is 48.

**To Reproduce**
Try to build memcached with gcc-12.0.1 and use -Werror=array-bounds flag.

**System Information**
 - OS/Distro: Fedora
 - Version of OS/distro: Rawhide/36
 - Version of memcached: 1.6.13
 - Hardware detail: Every architecture

**Comment by dormando** [comment]
Neat, thanks!

**Comment by dormando** [comment]
released in 1.6.14
